scripts/cleanup_failed_registrations.py

```diff
@@ -25,21 +27,24 @@ def find_failed_registrations():
     )
     return {node.root for node in [job.dst_node for job in jobs] if node}
 
-def remove_failed_registrations(dry_run=True):
-    init_app(set_backends=True, routes=False)
+
+def remove_failed_registrations(dry=True):
     count = 0
     failed = find_failed_registrations()
-    if not dry_run:
-        for f in failed:
-            logging.info('Cleaning {}'.format(f))
-            if not f.registered_from:
-                logging.info('Node {0} had registered_from == None'.format(f._id))
-                continue
-            if not f.archive_job:  # Be extra sure not to delete legacy registrations
-                continue
-            f.archive_job.status = ARCHIVER_FAILURE
-            f.archive_job.sent = True
-            f.archive_job.save()
+    for f in failed:
+        logging.info('Cleaning {}'.format(f))
+        if not f.registered_from:
+            logging.info('Node {0} had registered_from == None'.format(f._id))
+            continue
+        if not f.archive_job:  # Be extra sure not to delete legacy registrations
+            continue
+        f.archive_job.status = ARCHIVER_FAILURE
+        f.archive_job.sent = True
+        f.archive_job.save()
+        if dry:
+            # We want to avoid sending emails here, rather we call the delete tree method ourselves for validation.
+            delete_registration_tree(f)
```

Comment:
The main application execution is wrapped in a `TokuTransaction` which if `dry = True` it will purposefully raise an exception preventing any changes from being written to the database. This ensures we thoroughly test all code branches and globally protects us of the scenario where we forgot to add a `if dry:` statement by hand anywhere in the run.